src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5188,6 +5193,256 @@ private static List<CompletionResult> CompleteRequires(CompletionContext context
+                }
+                return result.Count > 0 ? result : null;
+            }
+
+            if (lineKeyword.Value.Equals("EXTERNALHELP", StringComparison.OrdinalIgnoreCase))
+            {
+                context.WordToComplete = keywordArgument.Value;
+                var result = new List<CompletionResult>(CompleteFilename(context, containerOnly: false, (new HashSet<string>() { ".xml" })));
+                return result.Count > 0 ? result : null;
+            }
+
+            return null;
+        }
+
+        private static readonly IReadOnlyDictionary<string, string> s_commentHelpKeywords = new SortedList<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "SYNOPSIS", "A brief description of the function or script. This keyword can be used only once in each topic." },
+            { "DESCRIPTION", "A detailed description of the function or script. This keyword can be used only once in each topic." },
+            { "PARAMETER", ".PARAMETER  <Parameter-Name>\nThe description of a parameter. Add a .PARAMETER keyword for each parameter in the function or script syntax." },
+            { "EXAMPLE", "A sample command that uses the function or script, optionally followed by sample output and a description. Repeat this keyword for each example." },
+            { "INPUTS", "The .NET types of objects that can be piped to the function or script. You can also include a description of the input objects." },
+            { "OUTPUTS", "The .NET type of the objects that the cmdlet returns. You can also include a description of the returned objects." },
+            { "NOTES", "Additional information about the function or script." },
+            { "LINK", "The name of a related topic. Repeat the .LINK keyword for each related topic. The .Link keyword content can also include a URI to an online version of the same help topic." },
+            { "COMPONENT", "The name of the technology or feature that the function or script uses, or to which it is related." },
+            { "ROLE", "The name of the user role for the help topic." },
+            { "FUNCTIONALITY", "The keywords that describe the intended use of the function." },
+            { "FORWARDHELPTARGETNAME", ".FORWARDHELPTARGETNAME <Command-Name>\nRedirects to the help topic for the specified command." },
+            { "FORWARDHELPCATEGORY", ".FORWARDHELPCATEGORY <Category>\nSpecifies the help category of the item in .ForwardHelpTargetName" },
+            { "REMOTEHELPRUNSPACE", ".REMOTEHELPRUNSPACE <PSSession-variable>\nSpecifies a session that contains the help topic. Enter a variable that contains a PSSession object." },
+            { "EXTERNALHELP", ".EXTERNALHELP <XML Help File>\nThe .ExternalHelp keyword is required when a function or script is documented in XML files." }
+        };
+
+        private static readonly HashSet<string> s_commentHelpAllowedDuplicateKeywords = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
You might also find that this works. I don't think it's preferable to `var` in methods, but it's very convenient for object members.

```suggestion
        private static readonly HashSet<string> s_commentHelpAllowedDuplicateKeywords = new (StringComparer.OrdinalIgnoreCase)
```